async.md

```diff
@@ -0,0 +1,105 @@
+---
+---
+
+# A note on asynchronous code
+
+You might have noticed something in the small test we wrote in the introduction, namely the call to `await`. The Node.js engine is asynchronous, so when a call is made that takes some time to return (such as an API or Database call), rather than wasting time waiting for a response, the engine just processes the code that follows. This can result in issues when you try to use a variable that has not yet been initialized. For example, if you had:
```

Comment:
> The Node.js engine is asynchronous

It actually isn't. It's single threaded (unless you explicitly create child processes). You will only ever run one function at a time, and a function will run to completion. `async..await` is shorthand for creating a `Promise` that executes the function, using each `await` like a `.then`. An `async function` always returns a `Promise`, regardless of whether or not you `await` anything in it (so you'll need to `await` it).

Reply:
Also, `Promise`s are effectively a wrapper around `setTimeout` and friends, so nothing special beyond making them core features.